### PR TITLE
feat(typeahead highlighting): added configurable highlighting

### PR DESCRIPTION
**Pull request by barrycann**

### CODE REVIEW

File: projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts [hunk 1]
```diff
@@ -1,21 +1,34 @@
-import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Component, EventEmitter, Input, Output, Host} from '@angular/core';
+import {TypeaheadComponent} from '../typeahead.component';
```
Comment (tfoxley) [inline projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts]: Can this circular dependency now be removed?

File: projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
 
+    @Input()
+    typedChars: string;
+
     @Output()
     _selected: EventEmitter<any> = new EventEmitter<any>();
 
+    // Constructor(@Host() parent: TypeaheadComponent) {
+    //     console.log('this is parent: ', parent);
+    // }
```
Comment (tfoxley) [inline projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts]: Please delete this commented out code.

File: projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts [hunk 3]
```diff
@@ -1,21 +1,34 @@
-import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Component, EventEmitter, Input, Output, Host} from '@angular/core';
+import {TypeaheadComponent} from '../typeahead.component';
 
 @Component({
     selector: 'hc-typeahead-item',
     templateUrl: './typeahead-item.component.html',
-    styleUrls: ['./typeahead-item.component.scss']
+    styleUrls: ['./typeahead-item.component.scss'],
```
Comment (tfoxley) [inline projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts]: I think the linter will hate the trailing comma.

File: projects/cashmere/src/lib/typeahead/typeahead.component.ts [hunk 4]
```diff
@@ -207,6 +223,7 @@ export class TypeaheadComponent extends HcFormControlComponent implements OnInit
                 // and add highlighting to the new number 1.
                 this.setHighlighted(0, false);
                 this.valueChange.emit(value);
+                console.log(this._options);
```
Comment (tfoxley) [inline projects/cashmere/src/lib/typeahead/typeahead.component.ts]: Remove the console.log statement.

File: projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts (hunk 1 above)
Comment (barrycann) [reply projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts]: Change complete!

File: projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts (hunk 2 above)
Comment (barrycann) [reply projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts]: Change complete!

File: projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts (hunk 3 above)
Comment (barrycann) [reply projects/cashmere/src/lib/typeahead/typeahead-item/typeahead-item.component.ts]: Change complete!

File: projects/cashmere/src/lib/typeahead/typeahead.component.ts (hunk 4 above)
Comment (barrycann) [reply projects/cashmere/src/lib/typeahead/typeahead.component.ts]: Change complete!

**Pull request merged**